src/service/navigation.js

```diff
@@ -284,13 +284,10 @@ export class Navigation {
     }
 
     // Handle anchor transformations.
-    let am = this.anchorMutators_.peek();
-    while (am) {
-      this.anchorMutators_.dequeue();
-      am(target);
+    this.anchorMutators_.forEach(anchorMutator => {
```

Comment:
Tests that it can be done multiple times. 😉

Reply:
wrote test.